File: 02_Patterns/Pattern_21/Pattern_21.py
class Solution:
    def printTriangle(self, N):
        # outer loop for number of rows/lines
        for i in range(1, N+1):
            # inner loop for columns
            for j in range(1, N+1):
                # conditions when star will be printed
                if (i == 1 or i == N or j == 1 or j == N):
                    print("*", end="")
                else:
                    print(" ", end="")
            print()


# {
# Driver Code Starts
# Initial Template for Python 3
if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        N = int(input())
        ob = Solution()
        ob.printTriangle(N)
# } Driver Code Ends

File: 02_Patterns/Pattern_21/Pattern_21.js
function hollowSquarePattern(n) {
  // outer loop for number of rows/lines
  for (let i = 1; i <= n; i++) {
    let row = '';
    // inner loop for columns
    for (let j = 1; j <= n; j++) {
      // conditions when star will be printed
      if (i === 1 || i === n || j === 1 || j === n) {
        row += '*';
      } else {
        row += ' ';
      }
    }
    console.log(row);
  }
}

// call the function
// hollowSquarePattern(4);

// Export the function for testing
module.exports = hollowSquarePattern;
